packages/io/x3d-deserializer/src/translate.js

```diff
@@ -56,22 +56,9 @@ const codify = (x3d, data) => {
 // Units : ${length.name} (${length.factor})
 // Angles : ${angle.name} (${angle.factor})
 
-const {booleans, extrusions, geometries, maths, primitives, transforms, utils} = require('@jscad/modeling')
-const { colorize } =  require('@jscad/modeling').colors
-
-const applyTransform = (matrix, ...objects) => {
```

Comment:
i think this needs to be left in as the translateDefinitions() creates functions which call this special routine. X3D is pretty complex.

Reply:
Sorry, I had meant to comment about this change. I appreciate that you are reviewing carefully @z3dev  :-)

It appears to me that this `applyTransform` does _exactly_ the same thing as the `transform` operation. The `transform` operation also accepts a matrix and a list of objects, and applies the transformation to each of the objects, while preserving color. From `transform.js`:

```js
export const transform = (matrix, ...objects) => {
```

So it seems like we can just delete this `applyTransform` function from the generated script, and just use `transform`. That is why I also changed the function in `x3d-deserializer/src/translateDefinitions.js`:

https://github.com/jscad/OpenJSCAD.org/pull/1294/files/809f1ad9232ebd7f8b52ddeb231e409b9f688d9b#diff-a6901a2707f9e65bf11d963375a9f993fd19496573e5d5a46041e9a0d124a7cf

```diff
-  return applyTransform(matrix, objects)
+  return transform(matrix, objects)
```

This works since we are importing everything as `import * from '@jscad/modeling'`, which includes `transform` in the flattened modeling package.

I was just cleaning up some of the imports and it seemed unnecessary so I made the change. But let me know if you'd like to me to revert this change for this PR.

Reply:
Hmm... I think these are fine. I doubled checked the reason for applyTransform(). Basically, this was a work around for the support of colors on geometries. I believe the transform() will be sufficient as you mentioned. As a side, I added a new issue to test the IO libraries as part of V3.